[Docs] improve NAT and manual peering documentation (#2827)
liqotech/liqo: docs/usage/namespace-offloading.md

### Namespace mapping strategy

When a namespace is offloaded, Liqo needs to create a *twin* namespace on the clusters where the namespaces is extended.
The *namespace mapping strategy* defines the naming strategy used to create the remote namespaces, and can be configured through the `--namespace-mapping-strategy` flag.
The accepted values are:

> * **DefaultName** (default): to **prevent conflicts** on the target cluster, remote namespace names are generated as the concatenation of the local namespace name and the cluster name of the local cluster (e.g., *foo* could be mapped to *foo-lively-voice*).

Reviewer: ```suggestion
* **DefaultName** (default): to **prevent conflicts** on the target cluster, remote namespace names are generated as the concatenation of the local namespace name and the cluster name of the local cluster (e.g., *foo* could be mapped to *foo-name-of-the-local-cluster*).
```

* **EnforceSameName**: remote namespaces are named after the local cluster's namespace.

> This approach ensures **naming transparency**, which is required by certain applications, as well as guarantees that **cross-namespace DNS queries** referring to reflected services work out of the box (i.e., without adapting the target namespace name).

Reviewer: ```suggestion
This approach ensures **naming transparency**, which is required by certain applications (e.g., Istio), as well as guarantees that **cross-namespace DNS queries** referring to reflected services work out of the box (i.e., without adapting the target namespace name).
```

> Yet, it can lead to **conflicts** in case a namespace with the same name already exists inside the selected remote clusters, ultimately causing the remote namespace creation request to be rejected.

Reviewer: ```suggestion
However, it can lead to **conflicts** in case a namespace with the same name already exists inside the selected remote clusters, ultimately causing the remote namespace creation request to be rejected.
For instance, this mapping strategy prevents two consumer clusters to offload a namespace with the same name (e.g., `istio-system`) on the same provider cluster. 
```

## Pod offloading

The remote clusters are backed by a Liqo Virtual Node, which allows the vanilla Kubernetes scheduler to address the remote cluster as target for pod scheduling.
However, by default the Liqo virtual nodes have a [Taint](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/) applied to them, which prevents pods from being scheduled on them, unless a *namespace offloading* is enabled in the namespace where the pod is running.

You have two different ways to determine whether a pod can be scheduled on a virtual node (so on a remote cluster):

> * Defining a **pod offloading strategy** for the offloaded namespaces (default), which tells where the pods created on that namespace should be scheduled (whether in the local cluster, the remote clusters, or both letting the vanilla K8s scheduler decide).

Reviewer: I would suggest to add an example of a command that implements this strategy. The same comment applies also to the 'runtimeclass' sentence below.

Author: We have some example in the following sections, so I would proceed with your second proposal.

* Setting the Liqo **RuntimeClass** in the pod, in this case, the namespace offloading strategy is ignored, and the pod will be scheduled to the virtual nodes.

### Pod offloading strategy

The *pod offloading strategy* defines high-level constraints about pod scheduling, and can be configured through the `--pod-offloading-strategy` flag.
The accepted values are:

* **LocalAndRemote** (default): pods deployed in the local namespace can be scheduled **both onto local nodes and onto virtual nodes**, hence possibly offloaded to remote clusters. This will leave the Kubernetes scheduler to decide about the best placement, based on the available resources and the pod requirements. You can still influence the scheduler decision on which pods should be scheduled onto virtual nodes using the [standard Kubernetes mechanisms to assign Pods to Nodes](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/).
* **Local**: pods deployed in the local namespace are enforced to be scheduled onto **local nodes only**, hence never offloaded to remote clusters.

> The extension of a namespace, forcing at the same time all pods to be scheduled locally, enables the consumption of local services from the remote cluster, as shown in the [*service offloading* example](/examples/service-offloading).

Reviewer: I would remove this sentence, moving into a more general sentence below (after "*Remote*).
E.g.,

It is worth mentioning that, independently from the selected pod offloading strategy, services associated to them are propagated to the entire namespace (both locally and in the remote cluster), hence enabling the above pods to be consumed from anywhere in the Liqo domain, as shown in the [*service offloading* example](/examples/service-offloading).

Author: I agree that the statement you proposed is useful and it is worth to be mentioned.
The one you want to remove is about not about services but about pods scheduling, so I would keep it. The idea is saying: "even though with the `LocalAndRemote` policy, the decision about where to schedule pod is up to the K8s vanilla scheduler, you can still influence its decision, like you would do with any other nodes of the cluster."

Reviewer: I'm not sure I got your point.
The sentence I was referring to is the following:
`The extension of a namespace, forcing at the same time all pods to be scheduled locally, enables the consumption of local services from the remote cluster, as shown in the [*service offloading* example](/examples/service-offloading).`
This sentence is about service offloading, not about pod scheduling. I feel this should not stay here, as this should be already included in the new sentence we added at line L97.
